shepherd_bms/include/segment.h

```diff
@@ -1,7 +1,115 @@
+{
+    private:
+        const uint32_t TEMP_CONV[52] = 
+        {
+            33200, 31500, 29900, 28400, 26900, 25600, 24300, 23100, 21900, 20900, 
+            19900, 18900, 18000, 17100, 16300, 15500, 14800, 14100, 13500, 12900, 
+            12300, 11700, 11200, 10700, 10200, 9780, 9350, 8940, 8560, 8190, 7840, 
+            7510, 7190, 6890, 6610, 6340, 6080, 5830, 5590, 5370, 5150, 4950, 4750, 
+            4570, 4390, 4220, 4060, 3900, 3750, 3610, 3470, 0
+        };
+
+        const uint32_t VOLT_TEMP_CONV[57] = 
+        {
+            45140, 44890, 44640, 44370, 44060, 43800, 43510, 43210, 42900, 42560,
+            42240, 41900, 41550, 41170, 40820, 40450, 40040, 39650, 39220, 38810, 
+            38370, 37950, 37500, 37090, 36650, 36180, 35670, 35220, 34740, 34230,
+            33770, 33270, 32770, 32280, 31770, 31260, 30750, 30240, 29720, 29220, 
+            28710, 28200, 27680, 27160, 26660, 26140, 25650, 25130, 24650, 24150, 
+            23660, 23170, 22670, 22190, 21720, 21240, 0
+        };
+
+        int segmentData[1] = {0}; //placeholder for internal segment data type
+
+        uint8_t localConfig[NUM_CHIPS][6] = {};
+
+        uint8_t chipConfigurations[NUM_CHIPS][6];
+
+        void pullChipConfigurations();
+
+        void writeChipConfigurations();
+
+        void pullThermistors();
+
+        void pullVoltages();
+
+        uint8_t steinhartEst();
+
+        void configureCommRegisters_forI2C(uint8_t dataToWrite[][3], uint8_t commOutput[][6]);
+
+        void configureDischarge(uint8_t chip, uint16_t cells);
+
+        uint8_t* getChipConfigurations();
+
+        void setChipConfigurations(uint8_t config[][6]);
+
+    public:
+        SegmentInterface();
+
+        ~SegmentInterface();
+
+        /**
+         * @brief Pulls all cell data from the segments and returns all cell data
+         * 
+         * @todo make sure that retrieving cell data doesn't block code too much
+         * 
+         * @return int*
+         */
+        int* retrieveSegmentData(); //@todo int* is a placeholder for an actual data type for segment data
+
+        /**
+         * @brief Enables/disables balancing for all cells
+         * 
+         * @param balanceEnable 
+         */
+        void enableBalancing(bool balanceEnable);
+
+        /**
+         * @brief Enables/disables balancing for a specific cell
+         * 
+         * @param chipNum 
+         * @param cellNum 
+         * @param balanceEnable 
+         */
+        void enableBalancing(uint8_t chipNum, uint8_t cellNum, bool balanceEnable);
+
+        /**
+         * @brief Returns if a specific cell is balancing
+         * 
+         * @param chipNum 
+         * @return true 
+         * @return false 
+         */
+        bool isBalancing(uint8_t chipNum, uint8_t cellNum);
+
+        /**
+         * @brief Returns if any cells are balancing
+         * 
+         * @return true 
+         * @return false 
+         */
+        bool isBalancing();
```

Comment:
Should this be if balancing is allowed or if any cells are actively balancing? Technically there is a situation where balancing is allowable, but all the cells are close enough to each other that they don't need to balance.

Reply:
This returns true if any of the cells are balancing. Might not need it, but I figured it might be useful in the future. I figured that the "balancing allowed" functionality would be a higher level control logic and might be controlled by the state machine/other things like that